modules/authentication/reducer.js

```diff
@@ -1,89 +1,84 @@
+import Immutable from 'immutable';
 import actionTypes from './constants/actionTypes';
 
-const INITIAL_STATE = {
+const INITIAL_STATE = Immutable.fromJS({
   email: null,
   error: null,
   id: null,
   isActive: false,
   name: null,
   uid: null,
   tokenInfo: {}
-};
+});
 
 function authentication(state = INITIAL_STATE, action) {
   switch (action.type) {
   case actionTypes.CLEAR_HEADERS:
   case actionTypes.SIGN_OUT:
-    return {
-      ...state,
-      email: null,
-      id: null,
-      name: null,
-      uid: null,
-      tokenInfo: {}
-    };
+    return state.withMutations((map) => {
+      map.set('email', null);
+      map.set('error', null);
+      map.set('id', null);
+      map.set('isActive', false);
+      map.set('name', null);
+      map.set('uid', null);
+      map.set('tokenInfo', Immutable.fromJS({}));
+    });
```

Comment:
it looks like you've added `error` and `isActive` I'm assuming they were just missing before.